### PR TITLE
ci: elasticsearch: Upload script index map examples

### PR DESCRIPTION
**Pull request by golowanow**
Add several examples for `upload_test_results_es.py` script usage with ElasticSearch index map files for the following use cases:

 * Twister test results.

 * Twister test results with recordings.

 * Memory Footprint data (`twister-footprint.json`).

complements #76889

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

File: scripts/ci/es_upload/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+Execute the upload script once for the first time to create your index at the ElasticSearch
+server with the appropriate map file, for example:
+```bash
+python3 ./scripts/ci/upload_test_results_es.py --create-index \
+  --index YOUR_INDEX_NAME \
+  --map-file YOUR_INDEX_MAP.json
+```
+
+
+## Upload tranformations
```
Comment (stephanosio) [inline scripts/ci/es_upload/README.md:42]:
```suggestion
## Upload transformations
```

File: scripts/ci/es_upload/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+The upload script has several command line options to change `twister.json` data:
+exclude excess data, extract substrings by regular expressions, change data structure
+making it suitable for Kibana default visual components
+(see `--help` for more details):
+
+* `--exclude` removes excess testsuite properties not needed to store them
+  at Elasticsearch index.
+
+* `--transform` applies regexp group parsing rules to string propertites extracting
```
Comment (stephanosio) [inline scripts/ci/es_upload/README.md:52]:
```suggestion
* `--transform` applies regexp group parsing rules to string properties extracting
```

File: scripts/ci/es_upload/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  derived object properties.
+
+* `--flatten` changes testsuite data structure in regard of one of its list components:
+  either `testcases` or `recording`, so each item there becomes an independent data record
+  inheriting all other testsuite properties, whereas the children object's properties
+  are renamed with the parent object's name as a prefix:
+  `testcases_` or `recording_` respectively.
+  Only one testsuite property object can be flattened this way per index upload.
+  All other testsuite objects will be treated accorging to the index structure.
```
Comment (stephanosio) [inline scripts/ci/es_upload/README.md:61]:
```suggestion
  All other testsuite objects will be treated according to the index structure.
```

File: scripts/ci/es_upload/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
+# ElasticSearch index map examples
+
+This directory contains [ElasticSearch data store](https://github.com/elastic/elasticsearch)
+index map examples to use them by [upload_test_results_es.py](../upload_test_results_es.py)
```
Comment (stephanosio) [inline scripts/ci/es_upload/README.md:4]:
```suggestion
index map examples for use by [upload_test_results_es.py](../upload_test_results_es.py)
```

**Review by stephanosio — COMMENTED** [review]
Found some typos. Looks good otherwise.

**Review by nashif — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**